tck/src/test/java/io/smallrye/concurrency/tck/lifecycle/LifecycleExecuter.java

```diff
@@ -0,0 +1,92 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,  
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.smallrye.concurrency.tck.lifecycle;
+
+import java.lang.reflect.Method;
+
+import com.arjuna.ats.arjuna.common.arjPropertyManager;
+import com.arjuna.ats.jta.utils.JNDIManager;
+import org.jboss.arquillian.container.spi.event.container.AfterDeploy;
+import org.jboss.arquillian.container.spi.event.container.BeforeUnDeploy;
+import org.jboss.arquillian.core.api.annotation.Observes;
+import org.jboss.arquillian.test.spi.TestClass;
+import org.jnp.server.NamingBeanImpl;
+
+/**
+ * LifecycleExecuter
+ *
+ * @author <a href="mailto:aslak@redhat.com">Aslak Knutsen</a>
+ * @version $Revision: $
+ */
+public class LifecycleExecuter {
+   private static final String txnStoreLocation = "target/tx-object-store";
+   private NamingBeanImpl namingBean;
+
+   private void registerJTA() throws Exception {
+      // Start a JNDI server
+      namingBean = new NamingBeanImpl();
+      namingBean.start();
+
+      // Bind the JTA implementation to the correct JNDI contexts
+      JNDIManager.bindJTAImplementation();
+
+      // Set object store location
+      arjPropertyManager.getObjectStoreEnvironmentBean().setObjectStoreDir(txnStoreLocation);
+   }
+
+   private void unregisterJTA() throws Exception {
+      namingBean.stop();
+   }
+
+/*   public void executeBeforeDeploy(@Observes BeforeDeploy event, TestClass testClass) {
+      execute(testClass.getMethods(io.smallrye.concurrency.tck.lifecycle.api.BeforeDeploy.class));
+   }*/
+
+   public void executeAfterDeploy(@Observes AfterDeploy event, TestClass testClass) throws Exception {
+      registerJTA();
+      execute(testClass.getMethods(io.smallrye.concurrency.tck.lifecycle.api.AfterDeploy.class));
+   }
+   
+   public void executeBeforeUnDeploy(@Observes BeforeUnDeploy event, TestClass testClass) throws Exception {
+      unregisterJTA();
+      execute(testClass.getMethods(io.smallrye.concurrency.tck.lifecycle.api.BeforeUnDeploy.class));
+   }
+
+/*   public void executeAfterUnDeploy(@Observes AfterUnDeploy event, TestClass testClass) {
+      execute(testClass.getMethods(io.smallrye.concurrency.tck.lifecycle.api.AfterUnDeploy.class));
+   }*/
+
+   private void execute(Method[] methods)
+   {
+      if(methods == null)
+      {
+         return;
+      }
+      for(Method method : methods)
+      {
+         try
+         {
+            method.invoke(null);
+         }
+         catch (Exception e) 
+         {
+            throw new RuntimeException("Could not execute @BeforeDeploy method: " + method, e);
```

Comment:
Hardcoded `@BeforeDeploy` will lead to misleading exception messages :-)

Reply:
@Ladicek Well spotted, since the PR was merged I will raise another issue for this change.